eng/pipelines/libraries/run-test-job.yml

```diff
@@ -205,3 +292,81 @@ jobs:
+            condition: always()
+
+          - template: /eng/pipelines/common/upload-artifact-step.yml
+            parameters:
+              rootFolder: $(MergedMchFileLocation)
+              includeRootFolder: false
+              archiveType: $(archiveType)
+              tarCompression: $(tarCompression)
+              archiveExtension: $(archiveExtension)
+              artifactName: 'SuperPMI_Collection_$(SuperPmiCollectionName)_$(SuperPmiCollectionType)_$(osGroup)$(osSubgroup)_$(archType)_$(buildConfig)'
+              displayName: 'Upload artifacts SuperPMI $(SuperPmiCollectionName)-$(SuperPmiCollectionType) collection'
+              condition: always()
+
+          # Add authenticated pip feed
+          - task: PipAuthenticate@1
+            displayName: 'Pip Authenticate'
+            inputs:
+              artifactFeeds: public/dotnet-public-pypi
+              onlyAddExtraIndex: false
+            condition: always()
+
+          # Ensure the Python azure-storage-blob package is installed before doing the upload.
+          - script: $(PipScript) install --user --upgrade pip && $(PipScript) install --user azure.storage.blob==12.5.0 --force-reinstall
+            displayName: Upgrade Pip to latest and install azure-storage-blob Python package
+            condition: always()
+
+          - script: $(PythonScript) $(Build.SourcesDirectory)/src/coreclr/scripts/superpmi.py upload -log_level DEBUG -arch $(archType) -build_type $(buildConfig) -mch_files $(MergedMchFileLocation)$(SuperPmiCollectionName).$(SuperPmiCollectionType).$(MchFileTag).mch -core_root $(Build.SourcesDirectory)/artifacts/bin/coreclr/$(osGroup).x64.$(buildConfigUpper)
+            displayName: 'Upload SuperPMI $(SuperPmiCollectionName)-$(SuperPmiCollectionType) collection to Azure Storage'
+            condition: always()
+            env:
+              CLRJIT_AZ_KEY: $(clrjit_key1) # secret key stored as variable in pipeline
+
+          - task: CopyFiles@2
+            displayName: Copying superpmi.log of all partitions
+            inputs:
+              sourceFolder: '$(MchFilesLocation)'
+              contents: '**/$(SuperPmiCollectionName).$(SuperPmiCollectionType)*.log'
+              targetFolder: '$(SpmiLogsLocation)'
+            condition: always()
+
+          - task: PublishPipelineArtifact@1
+            displayName: Publish SuperPMI logs
+            inputs:
+              targetPath: $(SpmiLogsLocation)
+              artifactName: 'SuperPMI_Logs_$(SuperPmiCollectionName)_$(SuperPmiCollectionType)_$(osGroup)$(osSubgroup)_$(archType)_$(buildConfig)'
+            condition: always()
+
+        ########################################################################################################
+        #
+        # End of SuperPMI processing
+        #
+        ########################################################################################################
```

Comment:
These changes add a lot of complexity to the already non-trivial libraries yml files. Please consider moving this block into another YML and conditionally import it based on the PMI boolean.